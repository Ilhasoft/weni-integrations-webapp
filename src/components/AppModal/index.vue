<template>
  <unnnic-modal
    ref="unnnic-remove-modal"
    :showModal="showRemoveModal"
    :text="$t('apps.details.actions.remove.title')"
    scheme="feedback-red"
    modal-icon="alert-circle-1"
    @close="toggleRemoveModal"
  >
    <span slot="message" v-html="$t('apps.details.actions.remove.description')"></span>
    <unnnic-button
      ref="unnnic-remove-modal-close-button"
      slot="options"
      type="tertiary"
      @click="toggleRemoveModal"
      >{{ $t('general.Cancel') }}</unnnic-button
    >

    <LoadingButton
      ref="unnnic-remove-modal-navigate-button"
      slot="options"
      type="primary"
      :isLoading="loadingDeleteApp"
      :loadingText="$t('general.loading')"
      :text="$t('apps.details.actions.remove.remove')"
      @clicked="removeApp(currentRemoval.code, currentRemoval.uuid)"
    />
  </unnnic-modal>
</template>

<script>
  import LoadingButton from '@/components/LoadingButton/index.vue';

  export default {
    name: 'AppModal',
    components: {
      LoadingButton,
    },
    data() {
      return {
        showAddModal: false,
        showRemoveModal: false,
        currentRemoval: null,
      };
    },
    methods: {
      toggleRemoveModal(app = null) {
        this.currentRemoval = app;
        this.showRemoveModal = !this.showRemoveModal;
      },
    },
  };
</script>
