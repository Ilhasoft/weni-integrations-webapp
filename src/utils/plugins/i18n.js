import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
