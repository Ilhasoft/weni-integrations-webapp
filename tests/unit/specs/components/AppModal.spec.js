import AppModal from '../../../../src/components/AppModal/index.vue';
import { mount, createLocalVue } from '@vue/test-utils';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('AppModal', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(AppModal, {
      localVue,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicModal: true,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleRemoveModal', async () => {
    const spy = spyOn(wrapper.vm, 'toggleRemoveModal');
    expect(spy).not.toHaveBeenCalled();
    wrapper.vm.toggleRemoveModal();
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should set RemoveModal state as closed on modal close', async () => {
    const removeModalComponent = wrapper.findComponent({ ref: 'unnnic-remove-modal' });
    await removeModalComponent.vm.$emit('close');
    expect(wrapper.vm.showRemoveModal).toBeTruthy();
  });
});
