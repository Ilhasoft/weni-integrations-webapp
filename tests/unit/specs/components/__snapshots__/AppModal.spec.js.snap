// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal should be rendered properly 1`] = `
<unnnicmodal-stub text="some specific text" scheme="feedback-red" modal-icon="alert-circle-1"><span>some specific text</span>
  <unnnic-button type="tertiary">some specific text</unnnic-button>
  <loadingbutton type="primary" loadingtext="some specific text" text="some specific text"></loadingbutton>
</unnnicmodal-stub>
`;
